core/src/main/clojure/xtdb/operator/join.clj

```diff
@@ -160,7 +160,7 @@
 (defmethod lp/emit-expr :cross-join [join-expr args]
   (emit-cross-join (emit-join-children join-expr args)))
 
-(defn- build-phase [^ICursor build-cursor, ^IRelationMap rel-map, pushdown-blooms]
+(defn- build-phase [^ICursor build-cursor, ^IRelationMap rel-map, pushdown-blooms, ^HashSet iid-set]
```

Comment:
nit: we usually type-hint these as `^Set`, i.e. use the interface rather than the concrete impl